vizro-core/docs/pages/user-guides/extensions.md

```diff
@@ -0,0 +1,105 @@
+
+
+Vizro offers the ability to combine ease of use of low-code configuration, with the flexibility of a range of high-code extensions to expand your dashboard's functionality.
+
+
+## Vizro customizations
+
+Vizro custom functions can be used to extend the default functionality of Vizro. You can create Python functions for customizing charts, tables, dashboard components
+callbacks in the form of actions, and reactive HTML components - then plug them directly
+into the existing Vizro dashboard configuration
+
+### [Custom charts](custom-charts.md)
+
+You can create custom chart functions in Vizro by wrapping Plotly chart code inside a
+Vizro chart function wrapper, and then use the functions directly inside the Vizro dashboard configuration. This enables the creation of `plotly.graph_objects` charts with multiple traces, or `plotly_express` charts with post update customizations
+
+
+### [Custom tables](custom-tables.md)
+
+If the available arguments for the [`dash_ag_grid`][vizro.tables.dash_ag_grid] or [`dash_data_table`][vizro.tables.dash_data_table] models are insufficient, you can create a custom Dash AG Grid or Dash DataTable.
+
+
+### [Custom components](custom-components.md)
+
+You can create a custom component based on any dash-compatible component (for example, [dash-core-components](https://dash.plotly.com/dash-core-components),
+[dash-bootstrap-components](https://dash-bootstrap-components.opensource.faculty.ai/), [dash-html-components](https://github.com/plotly/dash/tree/dev/components/dash-html-components)).
+
+All Vizro's components are based on `Dash` and ship with a set of defaults that can be modified. If you would like to overwrite one of those defaults, or if you would like to use extra `args` or `kwargs` of those components, then this is the correct way to include those. You can use any existing attribute of any underlying [Dash component](https://dash.plotly.com/#open-source-component-libraries) with this method.
+
+
+### [Custom actions](custom-actions.md)
+
+If you want to use the [`Action`][vizro.models.Action] model to perform functions that are not available in the [pre-defined action functions][vizro.actions], you can create your own custom action.
+Like other [actions](actions.md), custom actions can also be added as an element inside the [actions chain](actions.md#chain-actions), and triggered with one of dashboard components.
+
+
+### [Custom figures](custom-figures.md)
+
+Custom figures are useful when you need a component that reacts to
+[filter](filters.md) and [parameter](parameters.md) controls.
+
+Vizro's [`Figure`][vizro.models.Figure] model accepts the `figure` argument, where you can enter _any_ custom figure function
+as described in the [how-to guide for figures](figure.md).
+
+
+## Dash customizations
+
+Since Vizro is built using Dash, it is possible to use Dash callbacks directly in any Vizro dashboard. This enables you to code beneath the Vizro layer and control Dash directly,
```

Comment:
Maybe add a suitable link to Dash callback docs here?

Reply:
Yes, I agree with Jo. Few users I interacted with didn't know they can add pure dash callbacks to Vizro, so we should highlight that in the title and link properly.